File: app/src/main/java/com/sysu/edu/todo/TitleAdapter.java
package com.sysu.edu.todo;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.textview.MaterialTextView;
import com.sysu.edu.databinding.ItemTitleBinding;

public class TitleAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
    public static final int[] headers = {com.google.android.material.R.style.TextAppearance_Material3_TitleMedium, com.google.android.material.R.style.TextAppearance_Material3_TitleLarge_Emphasized, com.google.android.material.R.style.TextAppearance_Material3_TitleLarge};
    String title = "";
    int n = 0;

    public TitleAdapter(String title) {
        setTitle(title);
    }

    public TitleAdapter(String title, int n) {
        setTitle(title, n);
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
        notifyItemInserted(0);
    }

    public void setTitle(String title, int n) {
        this.title = title;
        setHeader(n);
    }

    @NonNull
    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new RecyclerView.ViewHolder(ItemTitleBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false).getRoot()) {
        };
    }

    /**
     * 设置标题样式
     * 0: TextAppearance_Material3_TitleMedium
     * 1: TextAppearance_Material3_TitleLarge_Emphasized
     * 2: TextAppearance_Material3_TitleLarge
     *
     */
    public void setHeader(int n) {
        this.n = n;
        notifyItemChanged(0);
    }

    @Override
    public void onBindViewHolder(@NonNull RecyclerView.ViewHolder holder, int position) {
        MaterialTextView titleView = ItemTitleBinding.bind(holder.itemView).title;
        titleView.setText(title);
        titleView.setTextAppearance(0 <= n && n < headers.length ? headers[n] : 0);
    }

    @Override
    public int getItemCount() {
        return 1;
    }

    @Override
    public int getItemViewType(int position) {
        return 2;
    }
}

File: app/src/main/java/com/miyuyan/sysuer/todo/TitleAdapter.kt
package com.miyuyan.sysuer.todo

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.R
import com.miyuyan.sysuer.databinding.ItemTitleBinding

class TitleAdapter : RecyclerView.Adapter<RecyclerView.ViewHolder?> {
	@JvmField var title: String? = ""
	
	constructor(title: String?) {
		this.title = title
	}
	
	constructor(title: String?, n: Int) {
		setTitle(title, n)
	}
	
	fun setTitle(title: String?) {
		this.title = title
		notifyItemInserted(0)
	}
	
	fun setTitle(title: String?, n: Int) {
		this.title = title
		header = n
	}
	
	override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
		return object :
			RecyclerView.ViewHolder(ItemTitleBinding.inflate(LayoutInflater.from(parent.context), parent, false)
										.getRoot()) {}
	}
	
	override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
		ItemTitleBinding.bind(holder.itemView).title.apply {
			text = title
			setTextAppearance(if (0 <= header && header < headers.size) headers[header] else 0)
		}
	}
	
	override fun getItemCount(): Int {
		return 1
	}
	
	var header: Int = 0
		/**
		 * 设置标题样式
		 * 0: TextAppearance_Material3_TitleMedium
		 * 1: TextAppearance_Material3_TitleLarge_Emphasized
		 * 2: TextAppearance_Material3_TitleLarge
		 * 
		 */
		set(header) {
			field = header
			notifyItemChanged(0)
		}
	
	companion object {
		val headers: IntArray = intArrayOf(R.style.TextAppearance_Material3_TitleMedium, R.style.TextAppearance_Material3_TitleLarge_Emphasized, R.style.TextAppearance_Material3_TitleLarge)
	}
}
